crates/storage/db/src/tables/mod.rs

```diff
@@ -97,55 +94,96 @@ pub trait TableViewer<R> {
     fn view<T: Table>(&self) -> Result<R, Self::Error>;
 
     /// Operate on the dupsort table in a generic way.
+    ///
     /// By default, the `view` function is invoked unless overridden.
     fn view_dupsort<T: DupSort>(&self) -> Result<R, Self::Error> {
         self.view::<T>()
     }
 }
 
+/// Defines all the tables in the database.
 macro_rules! tables {
-    ([
-        (TableType::Table, [$($table:ident),*]),
-        (TableType::DupSort, [$($dupsort:ident),*])
-    ]) => {
-        #[derive(Debug, PartialEq, Copy, Clone)]
-        /// Default tables that should be present inside database.
-        pub enum Tables {
+    (@bool) => { false };
+    (@bool $($t:tt)+) => { true };
+
+    (@view $name:ident $v:ident) => { $v.view::<$name>() };
+    (@view $name:ident $v:ident $_subkey:ty) => { $v.view_dupsort::<$name>() };
+
+    ($( $(#[$attr:meta])* table $name:ident<Key = $key:ty, Value = $value:ty $(, SubKey = $subkey:ty)? $(,)?>; )*) => {
```

Comment:
do we need this `table` marker token here? I see it's repeated for every table below

Reply:
Looks nicer :D